ocamldebug.el

```diff
@@ -717,8 +718,48 @@ Obeying it means displaying in another window the specified file and line."
 
 ;;; Miscellaneous.
 
+(defun ocamldebug--read-from-file (file)
+  "Read FILE as a list of sexps. If invalid syntax, return nil and message the error."
+  (with-temp-buffer
+    (save-excursion
+      (insert "(\n")
+      (insert-file-contents file)
+      (goto-char (point-max))
+      (insert "\n)\n"))
+    (condition-case err
+        (read (current-buffer))
+      ((error err) (progn (message "Error reading file %S: %S" file err) nil)))))
+
+(defun ocamldebug--find-single-library (sexps)
+  "If list SEXPS has a single element whose `car' is \"library\", return it.
+Otherwise return `nil'."
+  (let ((libs (seq-filter (lambda (elt) (equal (car elt) 'library)) sexps)))
+    (and (null (cdr libs)) (car libs))))
```

Comment:
Not worth nitpicking.  They both have their advantages and disadvantages.  Since Emacs-28 the "best practice" would be to use `length=`, but I think here it doesn't matter very much.